packages/core/src/modules/credentials/CredentialsAPI.ts

```diff
@@ -0,0 +1,397 @@
+import { MessageSender } from '../../agent/MessageSender'
+import { createOutboundMessage } from '../../agent/helpers'
+import { ServiceDecorator } from '../../decorators/service/ServiceDecorator'
+import { AriesFrameworkError } from '../../error'
+import { unitTestLogger } from '../../logger'
+import { ConnectionService } from '../connections/services/ConnectionService'
+import { IndyHolderService, IndyIssuerService } from '../indy'
+import { IndyLedgerService } from '../ledger'
+import { MediationRecipientService } from '../routing'
+
+import { CredentialProtocolVersion } from './CredentialProtocolVersion'
+import { CredentialResponseCoordinator } from './CredentialResponseCoordinator'
+import { CredentialState } from './CredentialState'
+import { CredentialsModule } from './CredentialsModule'
+import { CredentialRepository } from './repository'
+import { V1CredentialService } from './v1/V1CredentialService'
+import { V1LegacyCredentialService } from './v1/V1LegacyCredentialService'
+import { CredentialRecordType, CredentialExchangeRecord } from './v2/CredentialExchangeRecord'
+import { CredentialRole } from './v2/CredentialRole'
+import { V2CredentialService } from './v2/V2CredentialService'
+
+export interface CredentialsAPI {
+  // Proposal methods
+  proposeCredential(credentialOptions: ProposeCredentialOptions): Promise<CredentialExchangeRecord>
+  acceptCredentialProposal(credentialOptions: AcceptProposalOptions): Promise<CredentialExchangeRecord>
+
+  // negotiateProposal(credentialOptions: NegotiateProposalOptions): Promise<CredentialExchangeRecord>
+
+  // Offer methods
+  // offerCredential(credentialOptions: OfferCredentialOptions): Promise<CredentialExchangeRecord>
+  acceptCredentialOffer(credentialOptions: AcceptOfferOptions): Promise<CredentialExchangeRecord>
+  // declineOffer(credentialRecordId: string): Promise<CredentialExchangeRecord>
+  // negotiateOffer(credentialOptions: NegotiateOfferOptions): Promise<CredentialExchangeRecord>
+
+  // // Request
+  // requestCredential(credentialOptions: RequestCredentialOptions): Promise<CredentialExchangeRecord>
+
+  // when the issuer accepts the request he issues the credential to the holder
+  // acceptRequest(credentialOptions: AcceptRequestOptions): Promise<CredentialExchangeRecord>
+
+  // // Credential
+  // acceptCredential(credentialRecordId: string): Promise<CredentialExchangeRecord>
+
+  // // Record Methods
+  // getAll(): Promise<CredentialExchangeRecord[]>
+  getById(credentialRecordId: string): Promise<CredentialRecord>
+  // findById(credentialRecordId: string): Promise<CredentialExchangeRecord | null>
+  // deleteById(credentialRecordId: string): Promise<void>
+  // findByQuery(query: Record<string, Tag | string[]>): Promise<CredentialExchangeRecord[]>
+}
+
+@scoped(Lifecycle.ContainerScoped)
+export class CredentialsAPI extends CredentialsModule implements CredentialsAPI {
+  private connService: ConnectionService
+  private msgSender: MessageSender
```

Comment:
This was done to distinguish the local objects (v2) from the ones passes up to the base class. I renamed them with a v2prefix. This can be changed later when the base class (CredentialsModule) is removed.